File: ecosystems/rubygems/test/formatting_residue_test.rb
require "minitest/autorun"
require "aitextcleaner/formatting_residue"

class FormattingResidueTest < Minitest::Test
  def test_cleans_copied_formatting_residue
    assert_equal(
      "Hello world...\n\nNext line",
      Aitextcleaner::FormattingResidue.clean("<p>Hello <span>world</span>…</p><div>Next line</div>")
    )
  end
end

File: ecosystems/clojars/test/formatting_residue_test.clj
(ns aitextcleaner.formatting-residue-test
  (:require [clojure.test :refer [deftest is run-tests]]
            [aitextcleaner.formatting-residue :refer [clean-formatting-residue]]))

(deftest cleans-copied-formatting-residue
  (is (= "Hello world...\n\nNext line"
         (clean-formatting-residue "<p>Hello <span>world</span>…</p><div>Next line</div>"))))

(when (= *file* (System/getProperty "babashka.file"))
  (System/exit (if (zero? (:fail (run-tests)) ) 0 1)))
